src/app/services/api/api.service.ts

```diff
@@ -0,0 +1,103 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { catchError, EMPTY, Observable, of } from 'rxjs';
+import { map } from 'rxjs/operators';
+import {
+	DownloadsPoint,
+	DownloadsRange,
+	GithubRepositoryContributor,
+	GithubRepositoryLanguages,
+	GithubRepositoryOverview,
+	Suggestion,
+} from '../../models';
+
+@Injectable({
+	providedIn: 'root',
+})
+export class ApiService {
+	private readonly NPM_REGISTRY_ENDPOINT = `https://api.npmjs.org`;
+	private readonly GITHUB_ENDPOINT = `https://api.github.com/repos`;
+	constructor(private readonly httpClient: HttpClient) {}
+
+	/**
+	 * @Returns top contributors for a github repository
+	 *
+	 * @repositoryName - name of the repository: angular/angular-cli , vuejs/vuex
+	 *
+	 * example: https://api.github.com/repos/vuejs/pinia/contributors
+	 */
+	getGithubPackageContributors(repositoryName: string): Observable<GithubRepositoryContributor[]> {
+		return this.httpClient
+			.get<GithubRepositoryContributor[]>(`${this.GITHUB_ENDPOINT}/${repositoryName}/contributors`)
+			.pipe(
+				catchError((error: unknown) => {
+					console.error(error);
+
+					return of([]);
+				})
+			);
+	}
+
+	/**
+	 * @Returns languages that the github repository is written
+	 *
+	 * @repositoryName - name of the repository: angular/angular-cli , vuejs/vuex
+	 *
+	 * example: https://api.github.com/repos/vuejs/pinia/languages
+	 */
+	getGithubPackageLanguages(repositoryName: string): Observable<GithubRepositoryLanguages> {
+		return this.httpClient.get<GithubRepositoryLanguages>(`${this.GITHUB_ENDPOINT}/${repositoryName}/languages`).pipe(
+			catchError((error: unknown) => {
+				console.error(error);
+
+				return EMPTY;
```

Comment:
EMPTY observable never emits:

```typescript
EMPTY.subscribe(() => console.log('this never emits'));
```

This would effectively be the same as just letting the Observable error. Because of this, I think it'd be better to return some "default" value, null, or throw the error

Reply:
updated at [65083c4](https://github.com/bitovi/opensource-analytics-dashboard/pull/11/commits/65083c476baa315e7967f73f8364a787786fabaf)